lib/monitoring/monitor.rb

```diff
@@ -1,6 +1,7 @@
 module Monitoring
   class Monitor
-    def initialize(checker: checker,notifier: notifier)
+    def initialize(checker: nil,notifier: nil)
+      raise "both checker and notifier are required" if checker.nil? || notifier.nil?
```

Comment:
In CheckResult we raise a specific ArgumentException if there's a problem with the initialize() parameters -- is there a reason to use RuntimeError here?